From 49ffdc8cd60180a8e741c9c40d6178cf6244fe3c Mon Sep 17 00:00:00 2001
From: Simon Horman <horms+renesas@verge.net.au>
Date: Tue, 18 Jun 2013 17:58:46 +0900
Subject: [PATCH 0632/1083] Revert "ARM: shmobile: r8a7779: Add DU support"

This reverts commit fb5b892d806e9f4e265a41224afe3707ae0881ec.

Conflicts:
	arch/arm/mach-shmobile/setup-r8a7779.c
(cherry picked from commit 60a58e3bd634a2bdfc97d5e6fd484e2784b5d69b)

Signed-off-by: Ryo Kataoka <ryo.kataoka.wt@renesas.com>
---
 arch/arm/mach-shmobile/clock-r8a7779.c        |  2 +-
 arch/arm/mach-shmobile/include/mach/r8a7779.h |  2 --
 arch/arm/mach-shmobile/setup-r8a7779.c        | 23 -----------------------
 3 files changed, 1 insertion(+), 26 deletions(-)

diff --git a/arch/arm/mach-shmobile/clock-r8a7779.c b/arch/arm/mach-shmobile/clock-r8a7779.c
index 1766cec..f88c07b 100644
--- a/arch/arm/mach-shmobile/clock-r8a7779.c
+++ b/arch/arm/mach-shmobile/clock-r8a7779.c
@@ -184,7 +184,7 @@ static struct clk_lookup lookups[] = {
 	CLKDEV_DEV_ID("sh_mobile_sdhi.1", &mstp_clks[MSTP322]), /* SDHI1 */
 	CLKDEV_DEV_ID("sh_mobile_sdhi.2", &mstp_clks[MSTP321]), /* SDHI2 */
 	CLKDEV_DEV_ID("sh_mobile_sdhi.3", &mstp_clks[MSTP320]), /* SDHI3 */
-	CLKDEV_DEV_ID("rcar-du", &mstp_clks[MSTP103]), /* DU */
+	CLKDEV_DEV_ID("rcar-du.0", &mstp_clks[MSTP103]), /* DU */
 };
 
 void __init r8a7779_clock_init(void)
diff --git a/arch/arm/mach-shmobile/include/mach/r8a7779.h b/arch/arm/mach-shmobile/include/mach/r8a7779.h
index effc73b..bb9a671 100644
--- a/arch/arm/mach-shmobile/include/mach/r8a7779.h
+++ b/arch/arm/mach-shmobile/include/mach/r8a7779.h
@@ -5,7 +5,6 @@
 #include <linux/pm_domain.h>
 
 struct platform_device;
-struct rcar_du_platform_data;
 
 struct r8a7779_pm_ch {
 	unsigned long chan_offs;
@@ -23,7 +22,6 @@ static inline struct r8a7779_pm_ch *to_r8a7779_ch(struct generic_pm_domain *d)
 	return &container_of(d, struct r8a7779_pm_domain, genpd)->ch;
 }
 
-extern void r8a7779_add_du_device(struct rcar_du_platform_data *pdata);
 extern int r8a7779_sysc_power_down(struct r8a7779_pm_ch *r8a7779_ch);
 extern int r8a7779_sysc_power_up(struct r8a7779_pm_ch *r8a7779_ch);
 
diff --git a/arch/arm/mach-shmobile/setup-r8a7779.c b/arch/arm/mach-shmobile/setup-r8a7779.c
index 76a4fba..96bd203 100644
--- a/arch/arm/mach-shmobile/setup-r8a7779.c
+++ b/arch/arm/mach-shmobile/setup-r8a7779.c
@@ -22,10 +22,8 @@
 #include <linux/interrupt.h>
 #include <linux/irq.h>
 #include <linux/platform_data/gpio-rcar.h>
-#include <linux/platform_data/rcar-du.h>
 #include <linux/platform_device.h>
 #include <linux/delay.h>
-#include <linux/dma-mapping.h>
 #include <linux/input.h>
 #include <linux/io.h>
 #include <linux/serial_sci.h>
@@ -400,12 +398,6 @@ static struct platform_device sata_device = {
 	},
 };
 
-/* DU */
-static struct resource du_resources[] = {
-	DEFINE_RES_MEM(0xfff80000, 0x40000),
-	DEFINE_RES_IRQ(gic_iid(0x3f)),
-};
-
 static struct platform_device *r8a7779_early_devices[] __initdata = {
 	&scif0_device,
 	&scif1_device,
@@ -444,21 +436,6 @@ void __init r8a7779_add_standard_devices(void)
 			    ARRAY_SIZE(r8a7779_late_devices));
 }
 
-void __init r8a7779_add_du_device(struct rcar_du_platform_data *pdata)
-{
-	struct platform_device_info info = {
-		.name = "rcar-du",
-		.id = -1,
-		.res = du_resources,
-		.num_res = ARRAY_SIZE(du_resources),
-		.data = pdata,
-		.size_data = sizeof(*pdata),
-		.dma_mask = DMA_BIT_MASK(32),
-	};
-
-	platform_device_register_full(&info);
-}
-
 /* do nothing for !CONFIG_SMP or !CONFIG_HAVE_TWD */
 void __init __weak r8a7779_register_twd(void) { }
 
-- 
1.8.3.2

